### PR TITLE
osmesa & vtk & llvm & vtk-offscreen & u3d & vtku3dexporter & vtku3dexporter-offscreen

### PR DESCRIPTION
**Pull request by Maxyme**

### CODE REVIEW

File: vtk-offscreen/manual_vtk_test.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# This allows the interactor to initalize itself. It has to be called before an event loop.
+iren.Initialize()
+
+# We'll zoom in a little by accessing the camera and invoking a "Zoom" method on it.
+ren.ResetCamera()
+ren.GetActiveCamera().Zoom(1.5)
+renWin.Render()
+
+# Start the event loop.
+iren.Start()
```
Comment (Korijn) [inline vtk-offscreen/manual_vtk_test.py]: I'm a little surprised this test script does interactive stuff and render windows... for an offscreen VTK build?

File: vtk-offscreen/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+requirements:
+  build:
+    - python
+    - cmake >=3.6
+    - ninja >=1.7
+    - osmesa >=17.0.1  # [linux]
+
+  run:
+    - python
+    - osmesa >=17.0.1  # [linux]
```
Comment (Korijn) [inline vtk-offscreen/meta.yaml]: I think you should actually put `skip: true  # [not linux]` under `build`, and remove the platform selectors for `osmesa`

File: vtk-offscreen/run_test.py [hunk 3]
```diff
@@ -3,16 +3,8 @@
...
-    sys.exit(0)
-
 # create a rendering window and renderer
 ren = vtk.vtkRenderer()
-
-# Uncheck to test with osPray
-#ren.SetPass(vtk.vtkOSPRayPass())
-
 render_window = vtk.vtkRenderWindow()
 render_window.SetOffScreenRendering(True)  # This will render offscreen
```
Comment (Korijn) [inline vtk-offscreen/run_test.py]: this is not necessary for a pure offscreen VTK build

File: vtku3dexporter-offscreen/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package:
+  name: vtku3dexporter
```
Comment (Korijn) [inline vtku3dexporter-offscreen/meta.yaml]: wrong name!

File: vtk-offscreen/manual_vtk_test.py (hunk 1 above)
Comment (Maxyme) [inline vtk-offscreen/manual_vtk_test.py]: Correct, this test is not relevant for the offscreen version. Will remove.

File: vtk-offscreen/meta.yaml (hunk 2 above)
Comment (Maxyme) [inline vtk-offscreen/meta.yaml]: Good one.

File: vtk-offscreen/run_test.py (hunk 3 above)
Comment (Maxyme) [inline vtk-offscreen/run_test.py]: You mean explicitly setting the offscreen mode? Will test and remove if it works without.

File: vtku3dexporter-offscreen/meta.yaml (hunk 4 above)
Comment (Maxyme) [inline vtku3dexporter-offscreen/meta.yaml]: yes, correct

File: vtk-offscreen/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+requirements:
+  build:
+    - python
+    - cmake >=3.6
+    - ninja >=1.7
+    - osmesa >=17.0.1
+
+  run:
+    - python
+    - osmesa >=17.0.1
```
Comment (Korijn) [inline vtk-offscreen/meta.yaml]: should be hard pinning, this is never going to be compatible with another build

Comment (Maxyme) [inline vtk-offscreen/meta.yaml]: In theory it should be compatible with any version of libosmesa.so. In practice, probably not so it's a good practice call. Will fix.

**Pull request merged**